src/coreclr/jit/compiler.h

```diff
@@ -8471,7 +8471,7 @@ class Compiler
 
     // Gets a register mask that represent the kill set for a helper call since
     // not all JIT Helper calls follow the standard ABI on the target architecture.
-    regMaskTP compHelperCallKillSet(CorInfoHelpFunc helper);
+    AllRegsMask compHelperCallKillSet(CorInfoHelpFunc helper);
```

Comment:
To return a const reference, I need to have a copy of `AllRegsMask_*` struct that can be returned. In current implementation, I just replaced all the `#define RBM_*` with `#define AllRegsMask_* = AllRegsMask(...)` in `target.h` which basically means whenever we refer `AllRegsMask_*` we are essentially creating a new struct in the corresponding method and returning it, which means I cannot return reference of the locally created struct. The solution would be to have `AllRegsMask_` globally and then use them at various places. The problem with having them globally is because `RBM_FLT_CALLEE_TRASH` is mapped to `get_RBM_FLT_CALLEE_TRASH()` which is on the `compiler` object. The reason it is on `compiler` object is because we update that mask depending on `canUseEvexEncoding()` which is determined by the `compSupportsISA`, whici is again on the `compiler` object. I am not sure why `compSupportsISA` needs to be on a `compiler` object instead of a global or static that stays same for the lifetime of the process. But that change can be made some other day. For this situation, I am thinking of creating various `AllRegsMask_*` on `compiler` object and then use them throughout, without having to create these struct every time they are needed, and is better than what I am doing currently. Hopefully, `compiler` object is around at all the places where various `AllRegsMask_*` is present. In future, if we decide to make `compSupportsISA` global, we can move all the `AllRegsMask_*` definitions at global scope.